app/src/main/java/org/dhis2/usescases/teiDashboard/dashboardfragments/tei_data/TEIDataFragment.java

```diff
@@ -198,18 +198,22 @@ public void setData(DashboardProgramModel nprogram) {
     @Override
     public void onActivityResult(int requestCode, int resultCode, Intent data) {
         super.onActivityResult(requestCode, resultCode, data);
-        if (requestCode == REQ_EVENT && resultCode == RESULT_OK) {
-            presenter.getTEIEvents();
-            if (data != null) {
-                lastModifiedEventUid = data.getStringExtra(Constants.EVENT_UID);
-                if (((TeiDashboardMobileActivity) context).getOrientation() != Configuration.ORIENTATION_LANDSCAPE)
-                    getSharedPreferences().edit().putString("COMPLETED_EVENT", lastModifiedEventUid).apply();
-                else {
-                    if (lastModifiedEventUid != null)
-                        presenter.displayGenerateEvent(lastModifiedEventUid);
+        if(resultCode == RESULT_OK) {
+            if (requestCode == REQ_EVENT) {
+                presenter.getTEIEvents();
+                if (data != null) {
+                    lastModifiedEventUid = data.getStringExtra(Constants.EVENT_UID);
+                    if (((TeiDashboardMobileActivity) context).getOrientation() != Configuration.ORIENTATION_LANDSCAPE)
+                        getSharedPreferences().edit().putString("COMPLETED_EVENT", lastModifiedEventUid).apply();
```

Comment:
This could be a fragment extension (((ANYActivity) context).getOrientation() != Configuration.ORIENTATION_LANDSCAPE) :D

Reply:
I know it was here before like this !

Reply:
Ok!